### PR TITLE
Add AWS Key Management API Support

### PR DESCRIPTION
**Pull request by zacharyfox**
For comment only - will be adding tests and refactoring out the copy / pasted retry code.

### CODE REVIEW

File: src/erlcloud_kms.erl [hunk 1]
```diff
@@ -0,0 +1,1060 @@
...
+      }.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       s3_host=Host,
+       s3_port=Port
```
Comment (motobob) [inline src/erlcloud_kms.erl]: two lines above should be 

``` erlang
kms_host = Host,
kms_port = Port
```

File: src/erlcloud_kms.erl [hunk 2]
```diff
@@ -0,0 +1,1060 @@
...
+          any().
+create_alias(AliasName, TargetKeyId) ->
+    create_alias(AliasName, TargetKeyId, default_config()).
+
+
+-spec create_alias/3 ::
+          (AliasName :: string(),
+           TargetKeyId :: string(),
+           Config :: aws_config()) ->
+          any().
```
Comment (motobob) [inline src/erlcloud_kms.erl]: i doubt `any()` spec has any value.`erlcloud_aws:request` has pretty limited set of returns

File: src/erlcloud_kms.erl [hunk 3]
```diff
@@ -0,0 +1,1060 @@
...
+generate_random(NumberOfBytes) ->
+    generate_random(NumberOfBytes, default_config()).
+
+
+-spec generate_random/2 ::
+          (NumberOfBytes :: non_neg_integer(),
+           Config :: aws_config()) ->
+          any().
+generate_random(NumberOfBytes, Config) ->
+    Json = [{<<"NumberOfBytes">>, NumberOfBytes}],
```
Comment (motobob) [inline src/erlcloud_kms.erl:571]: this is a good one...
@zacharyfox i wonder what is our take on checking the limits here. 
http://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateRandom.html clearly states that 1< N <= 1024 this 
Shall we add guards all around or let AWS just return 500?

File: src/erlcloud_kms.erl [hunk 4]
```diff
@@ -0,0 +1,1060 @@
...
+          any().
+list_keys(Options) ->
+    list_keys(Options, default_config()).
+
+
+-spec list_keys/2 ::
+          (Options :: pagination_opts(),
+           Config :: aws_config()) ->
+          any().
+list_keys(Options, Config) ->
```
Comment (motobob) [inline src/erlcloud_kms.erl:774]: would it make sense to have list_all_\* in the library?

File: src/erlcloud_kms.erl [hunk 5]
```diff
@@ -0,0 +1,1060 @@
...
+    [":", erlang:integer_to_list(Port)].
+
+
+dynamize_options(List) ->
+    dynamize_options(List, []).
+
+
+dynamize_options([{Key, Value}|T], Acc) ->
+    case dynamize_option_key(Key) of
+        undefined -> dynamize_options(T, Acc);
```
Comment (motobob) [inline src/erlcloud_kms.erl]: I would like the library to validate the inputs and not just throw away unknowns silently. Maybe throw an exception or return  `{error, wrong_opt}` here?

File: include/erlcloud_aws.hrl [hunk 6]
```diff
@@ -27,6 +27,10 @@
           kinesis_host="kinesis.us-east-1.amazonaws.com"::string(),
           kinesis_port=80::non_neg_integer(),
           kinesis_retry=fun erlcloud_kinesis_impl:retry/2::erlcloud_kinesis_impl:retry_fun(),
+          kms_scheme="https://"::string(),
+          kms_host="kms.us-east-1.amazonaws.com"::string(),
+          kms_port=80::non_neg_integer(),
+          kms_retry=fun erlcloud_kms_impl:retry/2::erlcloud_kms_impl:retry_fun(),
```
Comment (motobob) [inline include/erlcloud_aws.hrl]: https://github.com/alertlogic/erlcloud/pull/13/files#diff-d9bda54b3e3a2f222e12d368ffd0d4ffR1009 `erlcloud_retry:retry` is already used and there is not file `erlcloud_kms_impl` so lets fix this default too.

File: src/erlcloud_kms.erl (hunk 3 above)
Comment (zacharyfox) [inline src/erlcloud_kms.erl:571]: I think we should let the service do it. The end result is the same, but the other option results in more code to maintain.

File: src/erlcloud_kms.erl (hunk 5 above)
Comment (zacharyfox) [inline src/erlcloud_kms.erl]: I think actually I'd like to change this to simply convert to the camel cased string form of the atom and let the service decide what to do with invalid input - it seems like a waste of effort to do it all here, and again - then we have that much code more to maintain.

File: include/erlcloud_aws.hrl (hunk 6 above)
Comment (zacharyfox) [inline include/erlcloud_aws.hrl]: As far as I can tell, this isn't used at all. I'll just remove it.

File: src/erlcloud_kms.erl (hunk 5 above)
Comment (motobob) [inline src/erlcloud_kms.erl]: ok. agree

**Comment by zacharyfox** [comment]
@motobob This is ready to go as fas as I can tell.

File: src/erlcloud_kms.erl [hunk 7]
```diff
@@ -0,0 +1,1045 @@
...
+       secret_access_key=SecretAccessKey
+      }.
+
+-spec new(string(), string(), string()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       s3_host=Host
```
Comment (motobob) [inline src/erlcloud_kms.erl]: this one is kms_host

**Comment by motobob** [comment]
@zacharyfox  yes - good. please fix that one tine copy-paste lag and i'll merge it

**Pull request merged**